crossmap/1.0/convert_maf_coords.sh

```diff
@@ -0,0 +1,113 @@
+MODE="${4:-crossmap}"
+MAFCOLSEP="${MAFCOLSEP:-|}"
+DEBUG="${DEBUG:-0}"
+
+# For colour
+RED='\033[0;31m'
+NC='\033[0m'
+
+DATE=$(date --rfc-3339 seconds | cut -c1-19)
+echo "[${DATE}] INFO: Checking whether there are \"${MAFCOLSEP}\" characters in the input" \
+	"MAF file."
+if fgrep -q \'${MAFCOLSEP}\' "${INPUT_MAF}"; then
+	echo -e "${RED}[${DATE}] ERROR: ${INPUT_MAF} contains \"${MAFCOLSEP}\" characters, which" \
+		"are used to separate MAF columns in the intermediate BED files. Please set the" \
+		"MAFCOLSEP environment variable to change this.${NC}"; exit 1;
+fi
+
+DATE=$(date --rfc-3339 seconds | cut -c1-19)
+echo "[${DATE}] INFO: Creating temporary BED file (${INPUT_BED})" \
+	"from input MAF file."
+awk 'BEGIN {FS=OFS="\t"} \
+		{chrom = $5; start = $6; end = $7} \
+		$10 == "SNP" {start = start - 1; end = end} \
+		$10 == "DEL" {start = start - 1; end = end} \
+		$10 == "INS" {start = start; end = end - 1} \
+		$0 !~ /^(#|Hugo_Symbol)/ {row=gensub(/\t/, "'"${MAFCOLSEP}"'", "g", $0); print chrom, start, end, row}' \
+		"${INPUT_MAF}" | \
+	sed 's/ /_____/g' > "${INPUT_BED}"
+
+if [[ "${MODE}" == "crossmap" ]]; then
+	CROSSMAP="${CROSSMAP:-CrossMap.py}"
+	command -v "${CROSSMAP}" >/dev/null 2>&1 || { \
+		echo -e "${RED}[${DATE}] ERROR: ${CROSSMAP} is not in the PATH or the file doesn't" \
+			"exist.${NC}"; exit 1; }
+	DATE=$(date --rfc-3339 seconds | cut -c1-19)
+	echo "[${DATE}] INFO: Converting genomic coordinates using ${CROSSMAP}."
+	"${CROSSMAP}" bed "${CHAIN_FILE}" "${INPUT_BED}" "${OUTPUT_BED}"
+	mv -f "${OUTPUT_BED}.unmap" "${OUTPUT_UNMAPPED}"
+else
+	DATE=$(date --rfc-3339 seconds | cut -c1-19)
+	LIFTOVER="${LIFTOVER:-liftOver}"
+	command -v "${LIFTOVER}" >/dev/null 2>&1 || { \
+		echo -e "${RED}[${DATE}] ERROR: ${LIFTOVER} is not in the PATH or the file doesn't" \
+			"exist.${NC}"; exit 1; }
+	echo "[${DATE}] INFO: Converting genomic coordinates using ${LIFTOVER}." \
+	"${LIFTOVER}" "${INPUT_BED}" "${CHAIN_FILE}" \
+		"${OUTPUT_BED}" "${OUTPUT_UNMAPPED}"
+fi
+
+NUM_UNMAPPED=$(wc -l < "${OUTPUT_UNMAPPED}")
+if [[ ${NUM_UNMAPPED} > 0 ]]; then
+	DATE=$(date --rfc-3339 seconds | cut -c1-19)
+	echo -e "${RED}[${DATE}] WARNING: ${NUM_UNMAPPED} variants were not properly mapped to the" \
+		"target genomic coordinate system.${NC}"
+fi
+
+DATE=$(date --rfc-3339 seconds | cut -c1-19)
+echo "[${DATE}] INFO: Converting output BED file (${OUTPUT_BED}) to MAF format." \
+grep "^Hugo_Symbol" "${INPUT_MAF}" > "${OUTPUT_MAF}"
+awk 'BEGIN {FS=OFS="\t"} \
+		{row = gensub(/['"${MAFCOLSEP}"']/, "\t", "g", $4)} \
+		{print $1, $2, $3, row}' \
+		"${OUTPUT_BED}" \
+	| awk 'BEGIN {FS=OFS="\t"} \
+		{chrom = $1; start = $2; end = $3} \
+		$13 == "SNP" {start = start + 1; end = end} \
+		$13 == "DEL" {start = start + 1; end = end} \
+		$13 == "INS" {start = start; end = end + 1} \
+		{$8 = chrom; $9 = start; $10 = end; print $0}' \
+	| cut -f4- \
+	| awk 'BEGIN {FS=OFS="\t"} {$4 = "GRCh37"; print $0}' \
```

Comment:
I think the genome build should be not be hardcoded here. The one specified as the target build should be substituted.

Reply:
I tested both ways, and it worked in both directions. This line is hardcoding genome build into the output maf. I will fix it.

Reply:
That's what I meant. The output MAF needs to have the right build in it, not just always GRCh37.